From e42fa012b6c3e1ce8f7c8bdcaa87f768d8f4037e Mon Sep 17 00:00:00 2001
From: Craig Scott <craig.scott@crascit.com>
Date: Tue, 25 Jul 2017 22:23:55 +1000
Subject: [PATCH] Allow early access version trailing string to be mixed case

The original regex was expecting to match strings like "ea", but it has
been reported that other strings like "Debian" are also possible.
---
 Modules/FindJava.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Modules/FindJava.cmake b/Modules/FindJava.cmake
index 3d32560..d762452 100644
--- a/Modules/FindJava.cmake
+++ b/Modules/FindJava.cmake
@@ -143,7 +143,7 @@ if(Java_JAVA_EXECUTABLE)
       if(var MATCHES "java version \"([0-9]+\\.[0-9]+\\.[0-9_.]+.*)\"")
         # This is most likely Sun / OpenJDK, or maybe GCJ-java compat layer
         set(Java_VERSION_STRING "${CMAKE_MATCH_1}")
-      elseif(var MATCHES "openjdk version \"([0-9]+)-[a-z]+\"")
+      elseif(var MATCHES "openjdk version \"([0-9]+)-[A-Za-z]+\"")
         # OpenJDK 9 early access builds or locally built
         set(Java_VERSION_STRING "1.${CMAKE_MATCH_1}.0")
       elseif(var MATCHES "java full version \"kaffe-([0-9]+\\.[0-9]+\\.[0-9_]+)\"")
--
libgit2 0.25.0

